### ISSUE TITLE
Ability to detect & use variables in SQL Query

### ISSUE THREAD

**Issue by JacobWeyer** [jakob/Postico#339]
It'd be interesting if Postico could have a slide down menu that could detect variables in the sequel query. EX. if the query is SELECT * FROM table WHERE id = ${id} and it see's the ${id} in the string, it'd be able to show that as a variable we can change and that'd inject it into the query without having to change my variables to test sql queries when writing more complex ones.

**Comment by jakob** [comment]
Good idea, but what language is this? The usual placeholders are `?` (Rails) or `$1` (libpq) or `:id` (PHP PDO).

**Comment by JacobWeyer** [comment]
You're right, I'm using pg-promise library and the ${Id} is actually a
JavaScript placeholder.

**Comment by doekman** [comment]
This probably should work like [psql's variables](https://www.postgresql.org/docs/11/app-psql.html#APP-PSQL-VARIABLES), so you could also use special variables like `DBNAME`.

Basically the most imporant one, for queries like `GRANT CONNECT ON DATABASE :DBNAME TO my_new_role;`, so you don't need to hard-code the database name in SQL. I use the extension `.psql` for this, since this tool supports it.

**Comment by KurtPattyn** [comment]
The same functionality as in DBeaver would already suit me

**Comment by doekman** [comment]
The nice thing about **psql**'s variables is the [interpolation](https://www.postgresql.org/docs/9.2/app-psql.html#APP-PSQL-INTERPOLATION). One could write this:

```
SELECT ':custom_name' as "name", 
       ":custom_column" as "value" 
FROM some_table"`
```

And _psql_ will insert the value of the variable `custom_name` as string literal (single quotes), and `custom_column` as identifier literal (double quotes). Pretty handy...

**Comment by jakob** [comment]
Good point.

There's a small typo in your example: According to the docs, the syntax is  `:name`, `:'name'`, or `:"name"`

**Comment by doekman** [comment]
Good catch. Fixed.

**Comment by jakob** [comment]
Short update:

I've implemented query parameters. For details see: https://github.com/jakob/Postico/issues/297#issuecomment-3674903408

However, I implemented it in a different way than psql:

1) Postico always prompts the user for the values
2) Postico always quotes the parameter value (similar behavior that psql does when the user types `:'name'`)

I know this makes it incompatible with psql, but it's unlikely that psql scripts would work in Postico anyway. I think this feature is more targeted at users that want to copy-paste queries from their ORM.

Postico also does not yet support the `${id}` format, only `$id`. I think I could adjust my parser to support `${id}`, but  I'm not sure how popular that is?
